### PR TITLE
homework 7

### PR DESCRIPTION
**Pull request by psergeev**

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -60,13 +64,13 @@ export default class extends React.PureComponent<Props, State> {
 
         return (
             <main>
-                <Router>
+                <this.props.Router location={this.props.location}>
```
Comment (loginov-rocks) [inline src/components/App/App.tsx]: My eyes are bleeding.

File: src/components/HeaderWithDetails/HeaderWithDetails.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default class extends React.PureComponent<Props, {}> {
         this._handleBackToSearchClick = this._handleBackToSearchClick.bind(this);
     }
 
-    public componentDidMount() {
+    public componentWillMount() {
```
Comment (loginov-rocks) [inline src/components/HeaderWithDetails/HeaderWithDetails.tsx]: Deprecated https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

File: src/components/HeaderWithSearch/HeaderWithSearch.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ export default class extends React.PureComponent<Props, State> {
         this._onSearchButtonClick = this._onSearchButtonClick.bind(this);
     }
 
-    public componentDidMount() {
+    public componentWillMount() {
```
Comment (loginov-rocks) [inline src/components/HeaderWithSearch/HeaderWithSearch.tsx]: Deprecated https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

**Review by loginov-rocks — CHANGES_REQUESTED** [review]

**Comment by loginov-rocks** [comment]
Cypress broken?

Screen Shot 2019-04-09 at 4 16 17 PM

**Comment by loginov-rocks** [comment]
Fetching for http://localhost:8000/search/star%20wars results into:

```html
<div id="app"><main><section class="results"><div class="results-sorter"><div class="hidden"><div>0 movie(s) found</div><div><span>Sort by</span><a href="#" class="sort-by-button ">release date</a><a href="#" class="sort-by-button selected">rating</a></div></div></div></section><footer>netflixroulette</footer></main></div>
                <script>
                    window.PRELOADED_STATE = {"prevMovies":[],"movies":[],"isEmpty":true,"search":"","searchBy":"title","sortBy":"rating","selectedMovie":null}
                </script>
```

The same result when trying to fetch anything else, e.g.: http://localhost:8000/film/11

So it does not seem to make any requests from the server side, right?

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -22,12 +21,17 @@ interface Props {
...
     handleMovieCardClick: (movie: Movie) => void;
+    Router: any;
+    location: string;
 }
 
 interface State {
     hasError: boolean;
 }
 
+const AsyncHeaderWithSearch = loadable(() => import('../HeaderWithSearch'));
```
Comment (loginov-rocks) [inline src/components/App/App.tsx:32]: This part and the following line doesn't work as expected, please check loadable imports.

File: src/components/App/App.tsx (hunk 1 above)
Comment (psergeev) [reply src/components/App/App.tsx]: fixed

File: src/components/HeaderWithDetails/HeaderWithDetails.tsx (hunk 2 above)
Comment (psergeev) [reply src/components/HeaderWithDetails/HeaderWithDetails.tsx]: moved into constructor

File: src/components/HeaderWithSearch/HeaderWithSearch.tsx (hunk 3 above)
Comment (psergeev) [reply src/components/HeaderWithSearch/HeaderWithSearch.tsx]: moved into constructor

File: src/components/App/App.tsx (hunk 4 above)
Comment (psergeev) [reply src/components/App/App.tsx:32]: fixed

**Review by loginov-rocks — APPROVED** [review]
LGTM

**Pull request merged**